### ISSUE TITLE
#[derive(JsonSchema)] compile errors when fields are dynamically generated

### ISSUE THREAD

**Issue by up9cloud** [GREsau/schemars#66]
Something like this:

>  /tmp/b$ cat Cargo.toml 

```toml
[package]
name = "b"
version = "0.1.0"
authors = ["up9cloud <8325632+up9cloud@users.noreply.github.com>"]
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
schemars = "0.8"
```

> /tmp/b$ cat src/main.rs 

```rust
macro_rules! build_struct {
    (
        $id:ident { $($t:tt)* }
    ) => {
        #[derive(Debug, schemars::JsonSchema)]
        pub struct $id {
            x: u8,
            $($t)*
        }
    };
}
build_struct!(A { v: i32, });
fn main() {
   println!("{:?}", A { x: 0, v: 1 });
}
```

> /tmp/b$ cargo run

```console
   Compiling b v0.1.0 (/private/tmp/b)
error[E0425]: cannot find value `gen` in this scope
  --> src/main.rs:12:22
   |
12 | build_struct!(A { v: i32, });
   |                      ^^^ not found in this scope

error[E0425]: cannot find value `schema_object` in this scope
  --> src/main.rs:12:22
   |
12 | build_struct!(A { v: i32, });
   |                      ^^^ not found in this scope

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0425`.
error: could not compile `b`

To learn more, run the command again with --verbose.
```

If I remove `schemars::JsonSchema` from derive, it works without errors.

**Comment by pjankiewicz** [comment]
I have exactly the same problem. Did you find any solution?

**Comment by up9cloud** [comment]
Unfortunately, no. If there is any solution, please let me know 🥲

**Comment by GREsau** [comment]
A fix for this has been merged and will be released in the next version 🙂

**Comment by GREsau** [comment]
Fixed in v0.8.3 - https://crates.io/crates/schemars/0.8.3
